extensions/hibernate-orm/deployment/src/main/resources/dev-ui/hibernate-orm-hql-console.js

```diff
@@ -32,120 +35,247 @@ export class HibernateOrmHqlConsoleComponent extends QwcHotReloadElement {
         }
 
-        .hidden {
-            visibility: hidden;
+        .json-preview {
+            color: var(--lumo-secondary-text-color);
+            cursor: pointer;
+            padding: 2px 4px;
+            border-radius: var(--lumo-border-radius-s);
+            width: fit-content;
         }
 
-        a, a:visited, a:focus, a:active {
-            text-decoration: none;
+        .json-preview:hover {
             color: var(--lumo-body-text-color);
         }
 
-        a:hover {
-            text-decoration: none;
-            color: var(--lumo-primary-text-color);
+        .table-container table {
+            border-collapse: collapse;
+            border-spacing: 0;
+            width: 100%;
         }
 
-        .font-large {
-            font-size: var(--lumo-font-size-l);
+        .table-container th {
+            text-align: left;
+            padding: 8px 12px;
+            font-weight: bold;
+            background-color: var(--lumo-contrast-5pct);
+            border-bottom: 1px solid var(--lumo-contrast-20pct);
+            white-space: nowrap;
         }
 
-        .cursor-text {
-            cursor: text;
+        .table-container td {
+            padding: 8px 12px;
+            vertical-align: top;
         }
 
-        .no-margin {
-            margin: 0;
+        .table-container tbody tr:nth-child(even) {
+            background-color: var(--lumo-contrast-5pct);
+        }
+
+        .nested-table {
+            width: 100%;
+            margin-top: 8px;
+            border-left: 2px solid var(--lumo-contrast-10pct);
+            padding: 0 10px;
+        }
+        
+        .nested-table table {
+            width: 100%;
+            border-collapse: collapse;
         }
-    `;
 
+        .nested-table th {
+            text-align: left;
+            padding: 4px 8px;
+            font-weight: bold;
+            background-color: var(--lumo-contrast-5pct);
+            border-bottom: 1px solid var(--lumo-contrast-20pct);
+        }
+
+        .nested-table td {
+            padding: 4px 8px;
+            vertical-align: top;
+        }
+
+        .vertical-separator {
+            width: 1px;
+            height: 24px;
+            background-color: var(--lumo-contrast-20pct);
+            margin: 0 8px;
+            align-self: center;
+        }
+    `;
 
     static properties = {
         _persistenceUnits: {state: true, type: Array},
         _selectedPersistenceUnit: {state: true},
-        _selectedEntity: {state: true},
-        _selectedEntityIndex: {state: true},
-        _currentHQL: {state: true},
-        _currentDataSet: {state: true},
-        _currentMessage: {state: true},
+        _entityTypes: {state: true, type: Array},
+        _messages: {state: true, type: Array},
         _currentPageNumber: {state: true},
         _currentNumberOfPages: {state: true},
         _allowHql: {state: true},
+        _assistantEnabled: {state: true},
+        _assistantInteractive: {state: true},
```

Comment:
Big +1 to Guillaume's suggestion to persist things like these two... Right now, if I want to take advantage of this feature, I have to click two/three buttons every time I leave the page and come back :/
I guess persisting the chat history would be complicated, but these booleans could be easy to store in a fashion similar to the "current theme" switch in the top right corner of the dev UI?

Note it can totally be a follow-up if we're in a hurry, but we'd need an issue to track such ideas.

Reply:
I would suggest local storage for that. See https://quarkus.io/guides/dev-ui#storage

Reply:
W.r.t giving the assistant memory, that is something we will have to do in Chappie

Reply:
> I would suggest local storage for that. See https://quarkus.io/guides/dev-ui#storage

Thanks once again @phillip-kruger, done.